packages/alfa-rules/src/helpers/element-checker.ts

```diff
@@ -0,0 +1,142 @@
+import { getRole, Role } from "@siteimprove/alfa-aria";
+import { BrowserSpecific } from "@siteimprove/alfa-compatibility";
+import { Device } from "@siteimprove/alfa-device";
+import {
+  // Element,
+  getElementNamespace,
+  getInputType,
+  InputType,
+  isElement,
+  Namespace,
+  Node
+} from "@siteimprove/alfa-dom";
+import { Option } from "@siteimprove/alfa-util";
+// import { Predicate } from "@siteimprove/alfa-util";
+
+function emptyOrHas<T>(elt: T, set?: Set<T>) {
+  return set === undefined || set.has(elt);
+}
+
+export class ElementChecker {
+  constructor(namesSet?: Set<string>, inputTypeSet?: Set<InputType>) {

```

Comment:
Having the extra variables here is only used for the class inheritance, making this a bit hacky. Normal use is the `with*` methods, even if initialisation could override this…